src/vulkan_as_2.cpp

```diff
@@ -115,38 +116,17 @@ Buffer prepare_buffer(const vulkan_setup_t& vulkan, VkDeviceSize size, void *dat
 	return buffer;
 }
 
-VkDeviceAddress get_buffer_device_adress(VkDevice device, VkBuffer buffer)
+VkDeviceAddress get_buffer_device_adress(const vulkan_setup_t& vulkan, VkBuffer buffer)
```

Comment:
I think this function too could be put into common code.

Reply:
I agree. I squashed the commits, and put the requested functionality into the common code, under `acceleration_structures` namespace.